crates/tools/src/prepare_scripts.rs

```diff
@@ -0,0 +1,389 @@
+        let repo_name = self
+            .source
+            .components()
+            .next()
+            .expect("get repo name")
+            .as_os_str();
+        make_path(target_root_dir, vec![repo_name, script_name])
+    }
+}
+
+pub fn prepare_scripts(
+    mode: ScriptsBuildMode,
+    input_path: &Path,
+    repos_dir: &Path,
+    scripts_dir: &Path,
+    output_path: &Path,
+) -> Result<()> {
+    let scripts_build_config = read_script_build_config(input_path);
+    match mode {
+        ScriptsBuildMode::Build => {
+            prepare_scripts_in_build_mode(&scripts_build_config, repos_dir, scripts_dir);
+        }
+        ScriptsBuildMode::Copy => {
+            prepare_scripts_in_copy_mode(&scripts_build_config.prebuild_image, scripts_dir);
+        }
+    }
+    check_scripts(&scripts_dir, &scripts_build_config.scripts);
+    generate_script_deploy_config(scripts_dir, &scripts_build_config.scripts, output_path)
+}
+
+fn read_script_build_config<P: AsRef<Path>>(input_path: P) -> ScriptsBuildConfig {
+    let input = fs::read_to_string(input_path).expect("read config file");
+    let mut scripts_build_config: ScriptsBuildConfig =
+        serde_json::from_str(&input).expect("parse scripts build config");
+    let default_build_config: ScriptsBuildConfig =
+        serde_json::from_str(&DEFAULT_BUILD_CONFIG).expect("parse scripts build config");
+    default_build_config.scripts.iter().for_each(|(k, v)| {
+        match scripts_build_config.scripts.get(k) {
+            Some(value) => {
+                let mut new = value.to_owned();
+                if PathBuf::default() == new.source {
+                    new.source.clone_from(&v.source);
+                }
+                new.always_success = value.always_success;
+                scripts_build_config.scripts.insert(k.to_owned(), new);
+            }
+            None => {
+                scripts_build_config
+                    .scripts
+                    .insert(k.to_owned(), v.to_owned());
+            }
+        }
+    });
+    scripts_build_config
+}
+
+fn prepare_scripts_in_build_mode(
+    scripts_build_config: &ScriptsBuildConfig,
+    repos_dir: &Path,
+    target_dir: &Path,
+) {
+    log::info!("Build scripts...");
+    run_pull_code(
+        scripts_build_config.repos.godwoken_scripts.clone(),
+        true,
+        repos_dir,
+        GODWOKEN_SCRIPTS,
+    );
+    run_pull_code(
+        scripts_build_config.repos.godwoken_polyjuice.clone(),
+        true,
+        repos_dir,
+        GODWOKEN_POLYJUICE,
+    );
+    run_pull_code(
+        scripts_build_config.repos.clerkb.clone(),
+        true,
+        repos_dir,
+        CLERKB,
+    );
+    build_godwoken_scripts(repos_dir, GODWOKEN_SCRIPTS);
+    build_godwoken_polyjuice(repos_dir, GODWOKEN_POLYJUICE);
+    build_clerkb(repos_dir, CLERKB);
+    copy_scripts_to_target(repos_dir, target_dir, &scripts_build_config.scripts);
```

Comment:
We either build the binaries or cp binaries from the docker image. Why do we do both at the same time?